src/js/view/floating/floating-drag-ui.js

```diff
@@ -9,52 +10,51 @@ export default class FloatingDragUI {
 
     disable() {
         if (this.ui) {
+            // Class is removed so the transition animation can fire again if the player re-floats.
+            removeClass(this.container, 'jw-floating-dragged');
             this.ui.destroy();
             this.ui = null;
         }
     }
 
     enable() {
-        let playerLeft;
-        let playerTop;
-        let innerHeight;
-        let innerWidth;
-        const auto = 'auto';
         const { container, input } = this;
-        const ui = this.ui = new UI(input, { preventScrolling: true })
-            .on('dragStart', () => {
-                playerLeft = container.offsetLeft;
-                playerTop = container.offsetTop;
-                innerHeight = window.innerHeight;
-                innerWidth = window.innerWidth;
+        let startX;
+        let startY;
+        let deltaX;
+        let deltaY;
+        let x = 0;
+        let y = 0;
+        // A min/max delta is assigned to prevent the player from being dragged off screen.
+        let minDeltaX;
+        let minDeltaY;
+        let maxDeltaX;
+        let maxDeltaY;
+        this.ui = new UI(input, { preventScrolling: true })
+            .on('dragStart', (e) => {
+                const { pageX, pageY } = e;
+                const { innerWidth, innerHeight } = window;
+                const { offsetLeft, offsetTop, offsetWidth, offsetHeight } = container;
+                startX = pageX;
+                startY = pageY;
+                minDeltaX = 0 - offsetLeft;
+                minDeltaY = 0 - offsetTop;
+                maxDeltaX = innerWidth - offsetLeft - offsetWidth;
+                maxDeltaY = innerHeight - offsetTop - offsetHeight;
+                // Class prevents initial animation styles from overriding translate styling.
+                addClass(container, 'jw-floating-dragged');
             })
             .on('drag', (e) => {
-                let left = Math.max(playerLeft + e.pageX - ui.startX, 0);
-                let top = Math.max(playerTop + e.pageY - ui.startY, 0);
-                let right = Math.max(innerWidth - (left + container.clientWidth), 0);
-                let bottom = Math.max(innerHeight - (top + container.clientHeight), 0);
-
-                if (right === 0) {
-                    left = auto;
-                } else {
-                    right = auto;
-                }
-                if (top === 0) {
-                    bottom = auto;
-                } else {
-                    top = auto;
-                }
-
+                const { pageX, pageY } = e;
+                deltaX = Math.max(Math.min(x + pageX - startX, maxDeltaX), minDeltaX);
+                deltaY = Math.max(Math.min(y + pageY - startY, maxDeltaY), minDeltaY);
                 style(container, {
-                    left,
-                    right,
-                    top,
-                    bottom,
-                    margin: 0
+                    transform: `translate(${deltaX}px, ${deltaY}px)`
```

Comment:
@zetagame @mamaddox I just remembered that we have a specific util for applying tranform styles that handles prefixes for old browsers. Please `transform(el, 'translate...` from 'utils/css 'rather than `style(el, { transform: 'translate...`.

Reply:
All of our supported browsers support this, not worth setting that many additional properties imo: 
https://caniuse.com/?search=transforms
https://support.jwplayer.com/articles/browser-support

Reply:
We can change the util when we have confirmed that. Use it so that this can be managed in the util rather than many places in the code.

Reply:
I just confirmed above that transform translate/translate3d do not need fallbacks in any of our supported browsers. Please check the links if unsure.

Reply:
OK. Update the util, and use it.

Reply:
Can do. To be clear there are other transform values that would benefit from the current util (such as backface-visibility), so there could be unintended side effects of modifying the util itself.

Reply:
@robwalch Done! Same performance :)